xdsl/irdl.py

```diff
@@ -755,16 +939,31 @@ def from_pyrdl(pyrdl_def: type[_OpT]) -> OpDef:
                 continue
             clsdict = {**clsdict, **parent_cls.__dict__}
 
-        type_hints = get_type_hints(pyrdl_def, include_extras=True)
-
         # Get all fields of the Operation class, including their parents classes
         opdict: dict[str, Any] = dict()
         for parent_cls in Operation.mro()[::-1]:
             opdict = {**opdict, **parent_cls.__dict__}
 
+        if "name" not in clsdict:
+            raise Exception(
+                f"pyrdl operation definition '{pyrdl_def.__name__}' does not "
+                "define the operation name. The operation name is defined by "
+                "adding a 'name' field."
+            )
+
+        type_var_mapping: dict[TypeVar, AttrConstraint] | None = None
+
+        # If the operation inherit from `Generic`, this means that it specializes a
+        # generic operation. Retrieve the mapping from `TypeVar` to pyrdl constraints.
+        if issubclass(pyrdl_def, Generic):
+            type_var_mapping = {
+                k: irdl_to_attr_constraint(v)
+                for k, v in get_type_var_mapping(pyrdl_def)[1].items()
+            }
+
         def wrong_field_exception(field_name: str) -> PyRDLOpDefinitionError:
             raise PyRDLOpDefinitionError(
-                f"{field_name} is neither a function, or an "
+                f"{pyrdl_def.__name__}.{field_name} is neither a function, or an "
```

Comment:
Makes the error slightly more useful, in my opinion, made it easier to track down bugs